app/Libraries/Mods.php

```diff
@@ -112,6 +112,19 @@ public function assertValidForMultiplayer(int $rulesetId, array $ids, bool $isRe
         }
     }
 
+    public function excludeModsAlwaysValidForSubmission(int $rulesetId, array $ids): array
+    {
+        $this->validateSelection($rulesetId, $ids);
+
+        $rulesetMods = $this->mods[$rulesetId];
+
+        return collect($ids)
+            ->filter(function ($id) use ($rulesetMods) {
+                return !$rulesetMods[$id]['AlwaysValidForSubmission'];
+            })
+            ->all();
```

Comment:
No need to use `collect` here. A plain `array_values(array_filter(` works.

Also `fn ($id) => $...` is simpler and doesn't need `use (...)`

Reply:
"idiot that doesn't write php tries to write php" moment. Will apply in a sec.